apps/civicsignalblog/src/payload/collections/Users.js

```diff
@@ -53,6 +56,76 @@ const Users = {
       },
       options: ROLE_OPTIONS,
     },
+    {
+      name: "defaultApp",
+      type: "select",
+      defaultValue: RESEARCH,
+      hasMany: false,
+      admin: {
+        isClearable: true,
+        isSortable: true,
+      },
+      options: applications,
+    },
+    {
+      name: "currentApp",
+      defaultValue: RESEARCH,
+      type: "select",
+      hasMany: false,
+      admin: {
+        isClearable: true,
+        isSortable: true,
+      },
+      options: applications,
+    },
+  ],
+  endpoints: [
+    {
+      path: "/current-app",
+      method: "patch",
+      handler: async (req, res) => {
+        if (!req.user) {
+          res.status(401).send({
+            error: "You need to be authenticated to perform this action",
+          });
+        }
+
+        const userId = req.user.id;
+
+        const { selectedApp } = req.body;
+
+        if (!selectedApp) {
+          res.status(400).send({
+            error: `Incorrect message format was received:${JSON.stringify(req.body)}`,
+          });
+        }
+
+        const currentUser = await payload.findByID({
+          collection: "users",
+          id: userId,
+          showHiddenFields: true,
+        });
+
+        if (!currentUser) {
+          res
+            .status(404)
+            .send({ error: "User with specified ID was not found" });
+        }
+
+        const updatedUser = await payload.update({
+          collection: "users",
+          id: userId,
+          data: {
+            currentApp: selectedApp || currentUser.defaultApp,
+          },
+        });
+
+        res.status(200).send({
```

Comment:
```suggestion
        }

        const { selectedApp } = req.body;
        if (!selectedApp) {
          res.status(400).send({
            error: `Incorrect message format was received:${JSON.stringify(req.body)}`,
          });
        }

        const userId = req.user.id;
        const currentUser = await payload.findByID({
          collection: "users",
          id: userId,
          showHiddenFields: true,
        });
        if (!currentUser) {
          res
            .status(404)
            .send({ error: "User with specified ID was not found" });
        }

        const updatedUser = await payload.update({
          collection: "users",
          id: userId,
          data: {
            currentApp: selectedApp || currentUser.defaultApp,
          },
        });
        res.status(200).send({
```

Too many empty lines make it harder to follow which parts of code are related/dependant.